internal/remediation/suggest/maven.go

```diff
@@ -52,23 +52,29 @@ func (ms *MavenSuggester) Suggest(ctx context.Context, client resolve.Client, mf
 			Type:       req.Type,
 			NewRequire: latest.Version,
 		}
+		origReq, depOrigin := originalRequirement(patch, specific.OriginalRequirements)
+		if origReq == "" {
+			// Cannot find the original requirement indicates the requirement not in base project
+			continue
+		}
 
-		origReq := originalRequirement(req, specific.RequirementsWithProperties)
-		if !requirementHasProperty(origReq) {
+		dep, _, _ := resolve.MavenDepTypeToDependency(patch.Type)
+		patch.Type = resolve.MavenDepType(dep, depOrigin)
```

Comment:
nit: maybe you can avoid calling `resolve.MavenDepTypeToDependency` twice if `originalRequirement()` just returns the `manifest.DependencyWithOrigin`?